### PR TITLE
Spec nominal ranges, nyquist, and behavior when outside the range.

### LINKED ISSUE #391: Specify nominal ranges for AudioParam
Some of them already have a nominal range, but some are missing. I also propose that we clamp the values to the nominal range if it happens to be outside the range.

The one that are missing, along with a proposal, are:
- GainNode.gain: `[-sampleRate/2; sampleRate/2]` (that is, the max of any other AudioParam, for AudioNode -> GainNode -> AudioParam purposes)
- AudioBufferSourceNode.playbackRate: `[0; 10000]`. The upper range is somewhat arbitrary

### PR DESCRIPTION
**Pull request by padenot**
This fixes #391.

### CODE REVIEW

**Comment by padenot** [comment]
How about something like this ?

**Comment by padenot** [comment]
Is this waiting for something ?

**Comment by joeberkovitz** [comment]
This also fixes #247

**Comment by rtoy** [comment]
What's the final conclusion here?  Add back the max and min attributes for every node that needs it? Perhaps they should be readonly so that the user can't do arbitrary things?

**Comment by joeberkovitz** [comment]
I propose that we begin by specing these in V1 and defer adding the discoverable stuff since we haven't discussed it in this cycle, and it may not be as easy to add back in as we think. We can't keep adding new features and this is a feature.

**Comment by rtoy** [comment]
It's not really a new feature; it's putting back what we used to have but ripped out. :-)

**Comment by joeberkovitz** [comment]
Well conservatively, putting on my "get this done" hat, that makes it a new feature... but if you and Paul feel you can implement this easily I will beat a retreat on this.

**Comment by rtoy** [comment]
Since Paul is the author, I'll let him decide what to do. I'm ok with either.

**Comment by joeberkovitz** [comment]
Resolution agreed by @rtoy and @padenot Merge this PR and make a new one with min and max discoverability.

**Comment by padenot** [comment]
This is rebased, squashed, and tidied.

File: index.html [hunk 1]
```diff
@@ -6168,29 +6203,30 @@ <h3 id="example-2">
...
+            0. It forms a <a>compound parameter</a> with <code>frequency</code>.
           </dd>
           <dt>
             readonly attribute AudioParam Q
           </dt>
           <dd>
             The <a href="https://en.wikipedia.org/wiki/Q_factor">Q</a> factor
-            has a default value of 1, with a nominal range of 0.0001 to 1000.
+            has a default value of 1. Its <a>nominal range</a> is [0.0001,
+            1000].
```
Comment (rtoy) [inline index.html]: Do we really need to limit the Q values to this range? From the Wikipedia article, I think we only need to limit Q to the range (0, inf).

Comment (padenot) [inline index.html]: Yes, I think I just did a mechanical conversion here.

**Comment by rtoy** [comment]
lgtm.

**Pull request merged**